.github/scripts/destroy-cluster.sh

```diff
@@ -11,4 +11,24 @@ kubectl get flcenvironments --namespace "$FLC_NAMESPACE"
 echo "Patching environment '$FLC_ENVIRONMENT' to 'not-deployed'"
 kubectl patch --namespace "$FLC_NAMESPACE" --type merge --patch '{"spec": {"desiredState": "environment-not-deployed"}}' flcenvironment "$FLC_ENVIRONMENT"
 
-kubectl wait --namespace "$FLC_NAMESPACE" --timeout="$DEFAULT_TIMEOUT" --for jsonpath='{.status.currentState}'=environment-not-deployed flcenvironment "$FLC_ENVIRONMENT"
+# wait until timeout or until the environment is in the desired state
+NEXT_WAIT_TIME=0
+while [[ $NEXT_WAIT_TIME -ne 20 ]]; do
+  # Check if the environment is in the desired state
+  current_state=$(kubectl get flcenvironment "$FLC_ENVIRONMENT" --namespace "$FLC_NAMESPACE" -ojsonpath='{.status.currentState}')
+
+  if [[ "$current_state" == "environment-not-deployed" ]]; then
+    echo "Environment '$FLC_ENVIRONMENT' has been destroyed successfully."
+    break
+  elif [[ "$current_state" == "environment-destruction-failed" ]]; then
+    echo "Environment '$FLC_ENVIRONMENT' destruction is failed. Please check the logs for more details."
+    exit 1
+  elif [[ -z "$current_state" ]]; then
+    echo "Environment '$FLC_ENVIRONMENT' does not exist or is not ready. Exiting."
+    exit 1
+  else
+    echo "Current state of environment '$FLC_ENVIRONMENT': '$current_state'. Waiting for desired state 'environment-not-deployed'..."
+    NEXT_WAIT_TIME=$((NEXT_WAIT_TIME+1))
+    sleep 60
+  fi
+done
```

Comment:
```
exit 1
``` 
should be added at the end of the script (line 35). Cluster is not deleted after 20 iteration means an error (like in line 25).

Reply:
fixed in 26c27ed9c